src/HealthChecks.Nats/NatsHealthCheck.cs

```diff
@@ -1,111 +1,38 @@
-using System.Text;
 using Microsoft.Extensions.Diagnostics.HealthChecks;
-using NATS.Client;
+using NATS.Client.Core;
 
 namespace HealthChecks.Nats;
 
 /// <summary>
 /// Health check for Nats Server.
 /// </summary>
-/// <remarks>
-/// Relies on a static <see cref="ConnectionFactory"/> which provides factory methods to create
-/// connections to NATS Servers, and a <see cref="IConnection"/> object connected to the NATS server.
-/// </remarks>
-public sealed class NatsHealthCheck : IHealthCheck, IDisposable
+public sealed class NatsHealthCheck(INatsConnection connection) : IHealthCheck
 {
-    private static readonly ConnectionFactory _connectionFactory = new();
-
-    private readonly NatsOptions _options;
+    public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, CancellationToken cancellationToken = default)
+    {
+        await TryConnectAsync(connection).ConfigureAwait(false);
```

Comment:
I meant just calling `ConnectAsync`. If it's in a healthy state, it just returns `Healthy`, if not, it throws and we return `Unhealthy`.

```suggestion
        return await TryConnectAsync(connection).ConfigureAwait(false);
```